### PR TITLE
Updated dependencies, removed PHP constraint and conflict

### LINKED ISSUE #5: [RFC]: Drop requirements and conflicts of subpackages in metapackage
## RFC

| Q                   | A      |
| ------------------- | ------ |
| Proposed Version(s) | 1.2.0  |
| BC Break?           | No |

### Goal

This goes in the same direction as laminas/laminas-mvc-plugins#5. Since this is a metapackage and there is no PHP code in this repository:

- [x] `php` should be dropped from the `require` section of `composer.json` (will be derrived by composer from the requirements of the subpackages)
- [x] `laminas/laminas-zendframework-bridge` should be dropped as well (it is included in the subpackages)
- [x] the `conflict` section should be droped from `composer.json` (should be and is part of `laminas-form`)

I can provide a PR if you agree with the changes.

### PR DESCRIPTION
**Pull request by driehle**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

Closes #5

### CODE REVIEW

**Comment by driehle** [comment]
@froschdesign Would you mind having a look at this?

**Comment by driehle** [comment]
@froschdesign Would you mind having a look at this?

File: composer.json [hunk 1]
```diff
@@ -25,14 +25,9 @@
...
-        "php": "^7.3 || ~8.0.0",
-        "laminas/laminas-code": "^3.5.0",
-        "laminas/laminas-form": "^2.16.3",
-        "laminas/laminas-i18n": "^2.11.1",
-        "laminas/laminas-zendframework-bridge": "^1.2.0"
-    },
-    "conflict": {
-        "laminas/laminas-code": "<3.5.0 || >=4.0.0"
+        "laminas/laminas-code": "^3.5.1",
+        "laminas/laminas-form": "^2.17.0",
```
Comment (froschdesign) [inline composer.json:29]: I think we can allow version 2 and 3:

```suggestion
        "laminas/laminas-form": "^2.17.0 || ^3.0.1",
```

Comment (driehle) [reply composer.json:29]: We could, not sure if we want though. Cause then we would keep the dependency to `laminas-code` which is not required with `laminas-form:^3.0` anymore. Additionally, we would still require the conflicts regarding `laminas-code`, which may bring trouble when other libraries are depending on `laminas-code` as well. To me that doesn't look promising and a new major with `laminas-form:^3.0` and without `laminas-code` would be more clean.

Comment (froschdesign) [reply composer.json:29]: You are absolutely right! Remove laminas-code and everything works as it should.

Comment (driehle) [reply composer.json:29]: Good, but that should still be a new major then, right? Cause `laminas-code` has been there before and enables annotation support for `laminas-form:^2.0`. Initially my idea was to suggest a 1.2.0 release with updated depencendies and forms ^2.0 and laminas-code, and a 2.0.0 with form ^3.0 and probably doctrine-annotations as well. Though doctrine-annotations are only required for annotations parsing that would be consistent with the 1.x releaes, which trough including laminas-code enabled annotation parsing right away. In that sense `laminas-mvc-form` would provide `laminas-form` including peer dependencies.

Comment (driehle) [reply composer.json:29]: @froschdesign push

Comment (Fahl-Design) [reply composer.json:29]: @froschdesign push

**Comment by Fahl-Design** [comment]
Hi, looks like there are some more updates that needs to be addressed: 

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - laminas/laminas-mvc-form[1.0.0, 1.0.x-dev] require php ^5.6 || ^7.0 -> your php version (8.0.9) does not satisfy that requirement.
    - laminas/laminas-mvc-form[1.1.0, 1.1.x-dev, 1.2.x-dev] require laminas/laminas-code ^3.5.0 -> found laminas/laminas-code[dev-develop, 3.5.0, 3.5.1, 3.5.x-dev (alias of dev-develop)] but it conflicts with your root composer.json require (^4.4.2).
    - Root composer.json requires laminas/laminas-mvc-form ^1.0 -> satisfiable by laminas/laminas-mvc-form[1.0.0, 1.0.x-dev, 1.1.0, 1.1.x-dev, 1.2.x-dev].

``` 

laminas-code is now at version 4.4.2

**Comment by driehle** [comment]
@Fahl-Design Well, with its v3 release, laminas-form switched its dependeny from laminas/laminas-code to doctrine/annotations, see https://docs.laminas.dev/laminas-form/v3/migration/v2-to-v3/#new-dependencies. The underlying reason is that laminas-code dropped its annotation parsing engine with their v4 release, so there is no way to update to laminas-code v4!

This is why from my point of view we must make two things with laminas-mvc-form (this metapackage here):
 - release 1.2.0 for laminas-form v2 and with laminas-code v3 as a peer dependency **(THIS MERGE REQUEST)**
 - release 2.0.0 for laminas-form v3 and with doctrine/annodations as a peer dependency _(yet to be done)_

@froschdesign @geerteltink @michalbundyra Would be great if one of you could have a look at this.

**Comment by driehle** [comment]
@froschdesign Is there still anything holding this back from being merged and released into 1.2.0?

**Comment by froschdesign** [comment]
@driehle 
Thank you for your time and this contribution! 👍 

_(And thanks for the reminder!)_

**Comment by driehle** [comment]
@froschdesign No worries and thanks for tagging a new release :-)

**Pull request merged**